File: CPP/Program/Garage.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include <vector>
#include "Car.cpp"
#include "Motorcycle.cpp"

// Deklarasi namespace
using namespace std;

// Kelas Garage yang digunakan untuk mempresentasikan data Garasi (Penyimpanan Kendaraan)
// Kelas Garage merupakan kelas mandiri
class Garage {
    private:
        // Atribut yang digunakan pada kelas Garage terdiri dari nama garasi, dan luas garasi.
        // Setiap Garasi memiliki mobil dan motor
        string nama_garasi;
        string luas_garasi;
        vector<Car> daftarMobil;
        vector<Motorcycle> daftarMotor;

    public:
        // Constructor
        Garage() {

        }

        // Constructor : Overloading
        Garage(string nama_garasi) {
            this->nama_garasi = nama_garasi;
            this->luas_garasi = "";
            this->daftarMobil = {};
            this->daftarMotor = {};
        }

        // Constructor : Overloading
        Garage(string nama_garasi, string luas_garasi) {
            this->nama_garasi = nama_garasi;
            this->luas_garasi = luas_garasi;
            this->daftarMobil = {};
            this->daftarMotor = {};
        }

        // Constructor : Overloading
        Garage(string nama_garasi, string luas_garasi, vector<Car> daftarMobil) {
            this->nama_garasi = nama_garasi;
            this->luas_garasi = luas_garasi;
            this->daftarMobil = daftarMobil;
            this->daftarMotor = {};
        }

        // Constructor : Overloading
        Garage(string nama_garasi, string luas_garasi, vector<Motorcycle> daftarMotor) {
            this->nama_garasi = nama_garasi;
            this->luas_garasi = luas_garasi;
            this->daftarMobil = {};
            this->daftarMotor = daftarMotor;
        }

        // Constructor : Overloading
        Garage(string nama_garasi, string luas_garasi, vector<Car> daftarMobil, vector<Motorcycle> daftarMotor) {
            this->nama_garasi = nama_garasi;
            this->luas_garasi = luas_garasi;
            this->daftarMobil = daftarMobil;
            this->daftarMotor = daftarMotor;
        }

        // Method setter untuk set nilai setiap atribut pada kelas Garage
        void setNamaGarasi(string nama_garasi) {
            this->nama_garasi = nama_garasi;
        }

        void setLuasGarasi(string luas_garasi) {
            this->luas_garasi = luas_garasi;
        }

        void setMobil(vector<Car> daftarMobil) {
            this->daftarMobil = daftarMobil;
        }

        void setMotor(vector<Motorcycle> daftarMotor) {
            this->daftarMotor = daftarMotor;
        }

        // Method getter untuk mendapatkan nilai setiap atribut pada kelas Garage
        string getNamaGarasi() {
            return nama_garasi;
        }

        string getLuasGarasi() {
            return luas_garasi;
        }

        vector<Car> getMobil() {
            return daftarMobil;
        }

        vector<Motorcycle> getMotor() {
            return daftarMotor;
        }

        // Destructor
        ~Garage() {

        }
};

File: Python/Program/Garage.py
# Mengimpor kelas Car dan Motorcycle
from Car import Car
from Motorcycle import Motorcycle

# Kelas Garage yang digunakan untuk mempresentasikan data Garasi (Penyimpanan Kendaraan)
# Kelas Garage merupakan kelas mandiri
class Garage:
    # Atribut yang digunakan pada kelas Garage terdiri dari nama garasi, dan luas garasi.
    # Setiap Garasi memiliki mobil dan motor
    
    # Constructor
    def __init__(self, nama_garasi="", luas_garasi="", daftarMobil=[], daftarMotor=[]):
        self.nama_garasi = nama_garasi
        self.luas_garasi = luas_garasi
        self.daftarMobil = daftarMobil
        self.daftarMotor = daftarMotor

    # Method setter untuk set nilai setiap atribut pada kelas Garage
    def setNamaGarasi(self, nama_garasi):
        self.nama_garasi = nama_garasi

    def setLuasGarasi(self, luas_garasi):
        self.luas_garasi = luas_garasi

    def setMobil(self, daftarMobil):
        self.daftarMobil = daftarMobil

    def setMotor(self, daftarMotor):
        self.daftarMotor = daftarMotor

    # Method getter untuk mendapatkan nilai setiap atribut pada kelas Garage
    def getNamaGarasi(self):
        return self.nama_garasi

    def getLuasGarasi(self):
        return self.luas_garasi

    def getMobil(self):
        return self.daftarMobil

    def getMotor(self):
        return self.daftarMotor
